### PR TITLE
Fixing paths related to `storages.journal`

### PR DESCRIPTION
**Pull request by eukaryo**
## Motivation

Related to PR https://github.com/optuna/optuna/pull/5555.

## Description of the changes

Fixed wrong paths of classes related to `JournalStorage`.

### CODE REVIEW

**Review by nabenabe0928 — APPROVED** [review]
Thank you for the PR, LGTM!

I also checked whether everything is covered by `git grep -l "<module_name>"`!

**Comment by not522** [comment]
LGTM. I'll merge after the conflicts are resolved.

File: docs/source/faq.rst [hunk 1]
```diff
@@ -556,7 +556,7 @@ If you want to use a file-based Optuna storage for these scenarios, please consi
    from optuna.storages import JournalStorage
    from optuna.storages.journal import JournalFileBackend
 
-   storage = JournalStorage(JournalFileStorage("optuna_journal_storage.log"))
+   storage = JournalStorage(JournalFileBackend("journal_file_storage_jsonl.log"))
```
Comment (not522) [inline docs/source/faq.rst:559]: Could you fix the file name?

```suggestion
   storage = JournalStorage(JournalFileBackend("optuna_journal_storage.log"))
```

**Review by not522 — APPROVED** [review]
LGTM

**Pull request merged**